### PR TITLE
In the case of Tuya end Moes, additional identification is carried out through the manufacturerName for(zigbee-OTA).

### PR DESCRIPTION
**Pull request by vladi1234**
https://github.com/Koenkk/zigbee2mqtt/issues/7941

https://github.com/Koenkk/zigbee-OTA/commit/4af6b2a99517f57c0111566a9735b40ecdbc2af7

### CODE REVIEW

**Comment by vladi1234** [comment]
Koenkk/zigbee2mqtt#7941

Koenkk/zigbee-OTA@4af6b2a

File: lib/ota/common.js [hunk 1]
```diff
@@ -430,6 +430,7 @@ async function getNewImage(current, logger, device, getImageMeta, downloadImage)
     assert(image.header.fileVersion === meta.fileVersion, 'File version mismatch');
     assert(!meta.fileSize || image.header.totalImageSize === meta.fileSize, 'Image size mismatch');
     assert(image.header.manufacturerCode === current.manufacturerCode, 'Manufacturer code mismatch');
+    assert(image.header.manufacturerName === current.manufacturerName, 'Manufacturer name mismatch');
```
Comment (Koenkk) [inline lib/ota/common.js:433]: This check should be removed,  the `image.header` never contains a `manufacturerName`

Comment (vladi1234) [reply lib/ota/common.js]: OK, everything is clear.

File: lib/ota/zigbeeOTA.js [hunk 2]
```diff
@@ -11,13 +11,15 @@ async function getImageMeta(current, logger, device) {
     const modelId = device.modelID;
     const imageType = current.imageType;
     const manufacturerCode = current.manufacturerCode;
+    const manufacturerName = current.manufacturerName;
```
Comment (Koenkk) [inline lib/ota/zigbeeOTA.js:14]: I think this should be `const manufacturerName = device.manufacturerName;`

Comment (vladi1234) [reply lib/ota/zigbeeOTA.js:14]: Yes

File: lib/ota/zigbeeOTA.js [hunk 3]
```diff
@@ -11,13 +11,15 @@ async function getImageMeta(current, logger, device) {
...
+    const manufacturerName = device.manufacturerName;
     const images = (await axios.get(url)).data;
 
     // NOTE: Officially an image can be determined with a combination of manufacturerCode and imageType.
     // However Gledopto pro products use the same imageType (0) for every device while the image is different.
     // For this case additional identification through the modelId is done.
+    // In the case of Tuya end Moes, additional identification is carried out through the manufacturerName.
     const image = images.find((i) => i.imageType === imageType && i.manufacturerCode === manufacturerCode &&
-        (!i.modelId || i.modelId === modelId));
+        (!i.modelId || i.modelId === modelId) && (!i.manufacturerName || i.manufacturerName === manufacturerName));
```
Comment (Koenkk) [inline lib/ota/zigbeeOTA.js:22]: This line should be changed to

```js
(!i.modelId || i.modelId === modelId) && (!i.manufacturerName || i.manufacturerName.includes(manufacturerName)));
```

After that, can you check if everything works, then I can merge it.

File: lib/ota/zigbeeOTA.js [hunk 4]
```diff
@@ -11,13 +11,15 @@ async function getImageMeta(current, logger, device) {
     const modelId = device.modelID;
     const imageType = current.imageType;
     const manufacturerCode = current.manufacturerCode;
+    const manufacturerName = device.manufacturerName;
     const images = (await axios.get(url)).data;
 
     // NOTE: Officially an image can be determined with a combination of manufacturerCode and imageType.
     // However Gledopto pro products use the same imageType (0) for every device while the image is different.
     // For this case additional identification through the modelId is done.
+    // In the case of Tuya end Moes, additional identification is carried out through the manufacturerName.
```
Comment (MattWestb) [inline lib/ota/zigbeeOTA.js:20]: // In the case of Tuya and Moes, additional identification is carried out through the manufacturerName.

Comment (vladi1234) [reply lib/ota/zigbeeOTA.js:20]: What is still missing?

Comment (MattWestb) [reply lib/ota/zigbeeOTA.js:20]: `Tuya end Moes` shall being `Tuya and Moes`.

End = somthing is stopping.
And = &

Comment (vladi1234) [reply lib/ota/zigbeeOTA.js:20]: Ah sory. Yes of course.

**Pull request merged**